packages/text-field/src/behaviors/TextInputBehavior.js

```diff
@@ -44,34 +59,45 @@ export default class TextInputBehavior extends Component {
     return this.state.value;
   }
 
+  /**
+   * @param {string} value
+   */
   setValue(value) {
     const { onChange } = this.props;
 
-    if (onChange) {
-      onChange(value);
+    if (onChange) onChange(value);
+
+    if (!this.isControlled()) {
+      this.setState({ value });
     }
+  }
 
-    this.setState({ value });
+  isControlled() {
+    return this.props.value !== undefined;
   }
 
+  /**
+   * @param {event} Event
+   */
   handleChange = event => {
-    this.setValue(this.isControlled() ? this.props.value : event.target.value);
+    this.setValue(event.target.value);
   };
 
+  /**
+   * An action that empties the value
+   */
   clear = () => {
     const { onClear } = this.props;
 
     if (onClear) {
       onClear();
     }
 
-    this.setValue("");
+    if (!this.isControlled()) {
+      this.setValue("");
```

Comment:
_To clarify:_

* When controlled...
    - `onClear` is called
    - `onChange` is **NOT** called
* When uncontrolled...
    - `onClear` is called
    - `onChange` is called